lib/net_buf/buf_simple.c

```diff
@@ -610,6 +637,16 @@ uint64_t net_buf_simple_pull_be64(struct net_buf_simple *buf)
 	return sys_be64_to_cpu(val);
 }
 
+char *net_buf_simple_to_str(struct net_buf_simple *buf)
+{
+	if (net_buf_simple_tailroom(buf) < 1) {
+		return NULL;
+	}
+
+	*net_buf_simple_tail(buf) = '\0';
+	return buf->data;
```

Comment:
If we have multiple buffer fragments, for example total 1kB len. Then this would only return pointer to first n. bytes. If we want to have something like this function, then user should provide a buffer where to copy the data. But we already have such a function called `net_buf_linearize()`

Reply:
Interesting! Do you mean it is better not provide a `to_str()` at the level of `net_buf_simple`?

A `net_buf_to_str()` function could call `net_buf_linearize()` to get a continuous region, and then `return net_buf_simple_to_str()` to add the '\0' and pass the `buf->data` pointer?

I can add more commits to that PR to include these.

Reply:
I just don't see the use case for an api that would copy buf to a string. If really needed one could use the linearize function and add null at the end if there is space.

Reply:
I throught that if these are to be used as too for growing strings in memory, they could be interleaved with functions from `string.h` and other APIs that rely on plain `char *`, and the `net_buf_simple_to_str()` would be there to facilitate this.

One example also [above](https://github.com/zephyrproject-rtos/zephyr/pull/79167#discussion_r1780053216) was strcmp():

```diff
 buf = message_received;

-str = net_buf_simple_to_str(buf);
-if (str == NULL) { 
-	return -ENOMEM;
-}
+if (net_buf_simple_tailroom(buf) < 1) {
+	return -ENOMEM;
+}
+net_buf_simple_add_u8(buf, 0);
+str = buf->data;

ret = validate_command(str);
if (ret < 0) {
	return ret;
}

+net_buf_simple_remove_mem(buf, 1);

if (net_buf_simple_headrooom(buf) < 1 || net_buf_simple_tailroom(buf) < 1) {
	return -ENOMEM;
}
net_buf_simple_push_u8(buf, '"');
net_buf_simple_add_u8(buf, '"');
```

Not a big difference!
It is mostly about having a wrapper that makes sure that the buffer can safely be used by `string.h` style functions. I also thought about cocci rules to check for `str...(buf->data...)` for instance.

Is this bloat? Easy for me to strip this function out of the tests and implementation! :)